### DISCUSSION TITLE
`Error: failed to update dependencies: Internal server error` happens when installing packages whose name includes `-`

### DISCUSSION THREAD

**Discussion by yudukikun5120** [deta/cloud-docs#269] [Help] [answered]
Runtime: Python 3.8

I got an error ``Error: failed to update dependencies: Internal server error``.

1. Add packages whose name includes '-' (e.g. _scikit-learn_ or _openap-top_ ) to `requirements.txt`.
2. Execute a command `deta deploy`.
3. Get the error:
```
❯ deta deploy      
Deploying...
Successfully deployed changes
Updating dependencies...
Error: failed to update dependencies: Internal server error
```

Is this a bug? Please notice me about this problem.

**Comment by jnsougata** [comment, marked as answer]
As `scikit-learn` `openap-top` `tensor flow` these libs are very big in size micros don't support this for now!

**Reply by yudukikun5120** [reply]
@jnsougata Thank you very much!

**Comment by abdelhai** [comment]
please check the docs https://docs.deta.sh/docs/micros/about#important-notes (7-8)
cc @aavshr any chance we could provide more context in the error message?

**Comment by jnsougata** [comment]
![image](https://user-images.githubusercontent.com/53375272/173238404-3f3a6f0b-5431-4c4b-a56f-e2a6796840a5.png)
